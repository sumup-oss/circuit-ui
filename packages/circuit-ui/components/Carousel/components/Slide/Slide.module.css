.base {
  position: relative;
  z-index: var(--slide-stack-order);
  flex-grow: 0;
  flex-shrink: 0;
  flex-basis: var(--slide-width);
  width: 100%;
  transform: translate3d(var(--slide-transform-x), 0, 0);
  backface-visibility: hidden;
}

.inner {
  overflow: hidden;
  will-change: width;
}

@keyframes slide-in {
  from {
    width: 0%;
  }

  to {
    width: 100%;
  }
}

@keyframes slide-out {
  from {
    width: 100%;
  }

  to {
    width: 0%;
  }
}

.animate-in {
  animation-name: slide-in;
  animation-duration: var(--slide-animation-duration);
  animation-fill-mode: forwards;
  animation-timing-function: var(--cui-transitions-slow);
}

.animate-out {
  animation-name: slide-out;
  animation-duration: var(--slide-animation-duration);
  animation-fill-mode: forwards;
  animation-timing-function: var(--cui-transitions-slow);
}

.content {
  width: var(--slide-width);
}
