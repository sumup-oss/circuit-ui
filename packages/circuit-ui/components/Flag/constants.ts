/**
 * Copyright 2025, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const COUNTRIES = [
  'AD',
  'AE',
  'AF',
  'AG',
  'AI',
  'AL',
  'AM',
  'AO',
  'AR',
  'AS',
  'AT',
  'AU',
  'AW',
  'AX',
  'AZ',
  'BA',
  'BB',
  'BD',
  'BE',
  'BF',
  'BG',
  'BH',
  'BI',
  'BJ',
  'BL',
  'BM',
  'BN',
  'BO',
  'BR',
  'BS',
  'BT',
  'BV',
  'BW',
  'BY',
  'BZ',
  'CA',
  'CC',
  'CD',
  'CF',
  'CG',
  'CH',
  'CI',
  'CK',
  'CL',
  'CM',
  'CN',
  'CO',
  'CR',
  'CU',
  'CV',
  'CW',
  'CX',
  'CY',
  'CZ',
  'DE',
  'DJ',
  'DK',
  'DM',
  'DO',
  'DZ',
  'EC',
  'EE',
  'EG',
  'EH',
  'ER',
  'ES',
  'ET',
  'FI',
  'FJ',
  'FK',
  'FM',
  'FO',
  'FR',
  'GA',
  'GB',
  'GD',
  'GE',
  'GF',
  'GG',
  'GH',
  'GI',
  'GL',
  'GM',
  'GN',
  'GP',
  'GQ',
  'GR',
  'GS',
  'GT',
  'GU',
  'GW',
  'GY',
  'HK',
  'HM',
  'HN',
  'HR',
  'HT',
  'HU',
  'ID',
  'IE',
  'IL',
  'IM',
  'IN',
  'IO',
  'IQ',
  'IR',
  'IS',
  'IT',
  'JE',
  'JM',
  'JO',
  'JP',
  'KE',
  'KG',
  'KH',
  'KI',
  'KM',
  'KN',
  'KP',
  'KR',
  'KW',
  'KY',
  'KZ',
  'LA',
  'LB',
  'LC',
  'LI',
  'LK',
  'LR',
  'LS',
  'LT',
  'LU',
  'LV',
  'LY',
  'MA',
  'MC',
  'MD',
  'ME',
  'MG',
  'MH',
  'MK',
  'ML',
  'MM',
  'MN',
  'MO',
  'MP',
  'MQ',
  'MR',
  'MS',
  'MT',
  'MU',
  'MV',
  'MW',
  'MX',
  'MY',
  'MZ',
  'NA',
  'NC',
  'NE',
  'NF',
  'NG',
  'NI',
  'NL',
  'NO',
  'NP',
  'NR',
  'NU',
  'NZ',
  'OM',
  'PA',
  'PE',
  'PF',
  'PG',
  'PH',
  'PK',
  'PL',
  'PM',
  'PN',
  'PR',
  'PS',
  'PT',
  'PW',
  'PY',
  'QA',
  'RE',
  'RO',
  'RS',
  'RU',
  'RW',
  'SA',
  'SB',
  'SC',
  'SD',
  'SE',
  'SG',
  'SH',
  'SI',
  'SJ',
  'SK',
  'SL',
  'SM',
  'SN',
  'SO',
  'SR',
  'SS',
  'ST',
  'SV',
  'SY',
  'SZ',
  'TC',
  'TD',
  'TF',
  'TG',
  'TH',
  'TJ',
  'TK',
  'TL',
  'TM',
  'TN',
  'TO',
  'TR',
  'TT',
  'TV',
  'TW',
  'TZ',
  'UA',
  'UG',
  'US',
  'UY',
  'UZ',
  'VA',
  'VC',
  'VE',
  'VG',
  'VI',
  'VN',
  'VU',
  'WF',
  'WS',
  'XK',
  'YE',
  'YT',
  'ZA',
  'ZM',
  'ZW',
] as const;

export const TERRITORIES = [
  'AQ',
  'BQ',
  'CP',
  'DG',
  'IC',
  'MF',
  'SX',
  'UM',
] as const;

export const SPECIAL_FLAGS = ['EU', 'UN', 'XX', 'PC'] as const;

export const FLAGS = [...COUNTRIES, ...TERRITORIES, ...SPECIAL_FLAGS] as const;
