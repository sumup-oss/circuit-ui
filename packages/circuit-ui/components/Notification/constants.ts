/**
 * Copyright 2023, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FC } from 'react';
import { Alert, Confirm, IconProps, Info, Notify } from '@sumup/icons';

export type NotificationVariant =
  | 'info'
  | 'success'
  | 'warning'
  | 'danger'
  /**
   * @deprecated
   */
  | 'confirm'
  /**
   * @deprecated
   */
  | 'notify'
  /**
   * @deprecated
   */
  | 'alert';

export const NOTIFICATION_ICONS: Record<
  NotificationVariant,
  FC<IconProps<'16' | '24'>>
> = {
  info: Info,
  success: Confirm,
  confirm: Confirm,
  warning: Notify,
  notify: Notify,
  danger: Alert,
  alert: Alert,
};

export const NOTIFICATION_COLORS: Record<
  NotificationVariant,
  { border: string; fg: string }
> = {
  info: {
    border: '--cui-border-accent',
    fg: '--cui-fg-accent',
  },
  success: {
    border: '--cui-border-success',
    fg: '--cui-fg-success',
  },
  confirm: {
    border: '--cui-border-success',
    fg: '--cui-fg-success',
  },
  warning: {
    border: '--cui-border-warning',
    fg: '--cui-fg-warning',
  },
  notify: {
    border: '--cui-border-warning',
    fg: '--cui-fg-warning',
  },
  danger: {
    border: '--cui-border-danger',
    fg: '--cui-fg-danger',
  },
  alert: {
    border: '--cui-border-danger',
    fg: '--cui-fg-danger',
  },
};

export const DEPRECATED_VARIANTS: Record<string, string> = {
  confirm: 'success',
  notify: 'warning',
  alert: 'danger',
};
