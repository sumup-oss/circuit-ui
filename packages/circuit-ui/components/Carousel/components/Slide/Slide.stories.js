/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import styled from '@emotion/styled';
import { css } from '@emotion/react';

import Headline from '../../../Headline';
import Image from '../../../Image';
import { spacing } from '../../../../styles/style-mixins';

import Slide from './Slide';

export default {
  title: 'Components/Carousel/Slide',
  component: Slide,
};

const headingStyles = css`
  color: #fff;
  width: 66%;
  position: absolute;
  bottom: 0;
  left: 25px;
  z-index: 2;
`;
const StyledHeadline = styled(Headline)(
  headingStyles,
  spacing({ bottom: 'giga' }),
);

export const OnlyImage = (args) => (
  <Slide {...args}>
    <Image
      src="/images/illustration-waves.jpg"
      alt="Aerial photo of turbulent blue ocean waves"
    />
  </Slide>
);

export const TextAndImage = (args) => (
  <Slide {...args}>
    <Image
      src="/images/illustration-waves.jpg"
      alt="Aerial photo of turbulent blue ocean waves"
    />
    <StyledHeadline>Get The SumUp Card Reader Today!</StyledHeadline>
  </Slide>
);
