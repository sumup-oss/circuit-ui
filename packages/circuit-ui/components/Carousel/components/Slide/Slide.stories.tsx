/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Headline } from '../../../Headline/index.js';
import { Image } from '../../../Image/index.js';

import { Slide, type SlideProps } from './Slide.js';

export default {
  title: 'Components/Carousel/Slide',
  component: Slide,
};

export const OnlyImage = (args: SlideProps) => (
  <Slide {...args}>
    <Image
      src="/images/sumup-coffee-transaction.jpg"
      alt="Next to a cup of coffee lays a phone showing a card transaction in the SumUp app"
    />
  </Slide>
);

export const TextAndImage = (args: SlideProps) => (
  <Slide {...args}>
    <Image
      src="/images/sumup-solo-cradle.jpg"
      alt="A SumUp Solo card reader sits in its charging cradle next to a tray of red, green and dark-brown mini tarts"
    />
    <Headline
      style={{
        color: '#fff',
        width: '66%',
        position: 'absolute',
        bottom: 0,
        left: '25px',
        zIndex: 2,
        marginBottom: 'var(--cui-spacings-giga)',
      }}
      as="h2"
    >
      Get The SumUp Card Reader Today!
    </Headline>
  </Slide>
);
