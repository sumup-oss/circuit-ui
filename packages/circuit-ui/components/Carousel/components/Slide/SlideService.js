/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { SLIDE_DIRECTIONS } from '../../constants';

export function getStackOrder(slideIndex, step, prevStep, slideDirection) {
  const isActive = step === slideIndex;
  const wasActive = prevStep === slideIndex;

  if (
    (slideDirection === SLIDE_DIRECTIONS.FORWARD && isActive) ||
    (slideDirection === SLIDE_DIRECTIONS.BACK && wasActive)
  ) {
    return 2;
  }

  if (
    (slideDirection === SLIDE_DIRECTIONS.FORWARD && wasActive) ||
    (slideDirection === SLIDE_DIRECTIONS.BACK && isActive)
  ) {
    return 1;
  }

  return slideIndex ? -slideIndex : 0;
}

export function shouldAnimate(slideIndex, step, prevStep, slideDirection) {
  const isActive = step === slideIndex;
  const wasActive = prevStep === slideIndex;

  if (slideDirection === SLIDE_DIRECTIONS.FORWARD) {
    return isActive;
  }

  if (slideDirection === SLIDE_DIRECTIONS.BACK) {
    return wasActive;
  }

  return false;
}

export function getDynamicWidth(width) {
  return width ? `${width}px` : '100%';
}
