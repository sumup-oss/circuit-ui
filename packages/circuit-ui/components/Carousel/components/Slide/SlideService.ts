/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { SlideDirection } from '../../constants.js';

export function getStackOrder(
  slideIndex?: number,
  step?: number,
  prevStep?: number,
  slideDirection?: SlideDirection,
) {
  const isActive = step === slideIndex;
  const wasActive = prevStep === slideIndex;

  if (
    (slideDirection === SlideDirection.FORWARD && isActive) ||
    (slideDirection === SlideDirection.BACK && wasActive)
  ) {
    return 2;
  }

  if (
    (slideDirection === SlideDirection.FORWARD && wasActive) ||
    (slideDirection === SlideDirection.BACK && isActive)
  ) {
    return 1;
  }

  return slideIndex ? -slideIndex : 0;
}

export function shouldAnimate(
  slideIndex?: number,
  step?: number,
  prevStep?: number,
  slideDirection?: SlideDirection,
) {
  const isActive = step === slideIndex;
  const wasActive = prevStep === slideIndex;

  if (slideDirection === SlideDirection.FORWARD) {
    return isActive;
  }

  if (slideDirection === SlideDirection.BACK) {
    return wasActive;
  }

  return false;
}

export function getDynamicWidth(width?: number) {
  return width ? `${width}px` : '100%';
}
