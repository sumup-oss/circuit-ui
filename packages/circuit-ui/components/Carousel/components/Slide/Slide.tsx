/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { HTMLAttributes, ReactNode } from 'react';

import { ANIMATION_DURATION, SLIDE_DIRECTIONS } from '../../constants.js';
import { clsx } from '../../../../styles/clsx.js';

import * as SlideService from './SlideService.js';
import classes from './Slide.module.css';

export interface SlideProps extends HTMLAttributes<HTMLDivElement> {
  /**
   *  Index of a slide in a carousel (required for animation).
   */
  index?: number;
  /**
   * Current active index of a carousel (required for animation).
   */
  step?: number;
  /**
   * Previous active index of a carousel (required for animation).
   */
  prevStep?: number;
  /**
   * Dimensions of a carousel (required for animation).
   */
  slideSize?: {
    width?: number;
    height?: number;
  };
  /**
   * Indicates slide direction of a carousel (required for animation).
   */
  slideDirection?: SLIDE_DIRECTIONS;
  /**
   * Indicates duration of animation between slides (in milliseconds)
   */
  animationDuration?: number;
  /**
   * Content of a slide
   */
  children: ReactNode;
}

export function Slide({
  index = 0,
  step = 0,
  prevStep,
  slideSize = {},
  slideDirection,
  animationDuration = ANIMATION_DURATION,
  children,
  ...props
}: SlideProps) {
  const stackOrder = SlideService.getStackOrder(
    index,
    step,
    prevStep,
    slideDirection,
  );
  const shouldAnimate = SlideService.shouldAnimate(
    index,
    step,
    prevStep,
    slideDirection,
  );
  const dynamicWidth = SlideService.getDynamicWidth(slideSize.width);
  const isAnimating = Boolean(slideSize.width && shouldAnimate);

  return (
    <div
      style={{
        '--slide-width': dynamicWidth,
        '--slide-stack-order': stackOrder,
        '--slide-transform-x': `${index * 100}%`,
        '--slide-animation-duration': `${animationDuration}ms`,
      }}
      className={classes.base}
      {...props}
    >
      <div
        className={clsx(
          classes.inner,
          isAnimating &&
            slideDirection === SLIDE_DIRECTIONS.FORWARD &&
            classes['animate-in'],
          isAnimating &&
            slideDirection === SLIDE_DIRECTIONS.FORWARD &&
            classes['animate-out'],
        )}
      >
        <div>{children}</div>
      </div>
    </div>
  );
}
