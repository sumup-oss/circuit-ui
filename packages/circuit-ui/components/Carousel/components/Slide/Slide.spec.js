/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { axe, render } from '../../../../util/test-utils';
import { SLIDE_DIRECTIONS } from '../../constants';

import Slide from './Slide';

describe('Slide', () => {
  describe('styles', () => {
    it('should render with default styles', () => {
      const { container } = render(<Slide />);

      expect(container).toMatchSnapshot();
    });

    it('should render with forward animation styles', () => {
      const { container } = render(
        <Slide
          index={0}
          step={0}
          slideSize={{ width: 800 }}
          slideDirection={SLIDE_DIRECTIONS.FORWARD}
        />,
      );

      expect(container).toMatchSnapshot();
    });

    it('should render with backward animation styles', () => {
      const { container } = render(
        <Slide
          index={0}
          prevStep={0}
          slideSize={{ width: 800 }}
          slideDirection={SLIDE_DIRECTIONS.BACK}
        />,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const { container } = render(<Slide />);
      const actual = await axe(container);

      expect(actual).toHaveNoViolations();
    });
  });
});
