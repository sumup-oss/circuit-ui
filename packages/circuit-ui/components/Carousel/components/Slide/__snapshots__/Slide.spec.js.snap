// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide styles should render with backward animation styles 1`] = `
@keyframes animation-0 {
  from {
    width: 100%;
  }

  to {
    width: 0%;
  }
}

.circuit-4 {
  width: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 800px;
  -ms-flex-preferred-size: 800px;
  flex-basis: 800px;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  position: relative;
  z-index: 2;
}

.circuit-0 {
  width: 800px;
}

.circuit-2 {
  box-shadow: 0 0 1px rgba(0,0,0,0.05);
  overflow: hidden;
  will-change: width;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-duration: 640ms;
  animation-duration: 640ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

<div
  class="circuit-4 circuit-5"
  width="800px"
>
  <div
    class="circuit-2 circuit-3"
  >
    <div
      class="circuit-0 circuit-1"
      width="800px"
    />
  </div>
</div>
`;

exports[`Slide styles should render with default styles 1`] = `
.circuit-4 {
  width: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  position: relative;
  z-index: 0;
}

.circuit-2 {
  box-shadow: 0 0 1px rgba(0,0,0,0.05);
  overflow: hidden;
  will-change: width;
}

.circuit-0 {
  width: 100%;
}

<div
  class="circuit-4 circuit-5"
  width="100%"
>
  <div
    class="circuit-2 circuit-3"
  >
    <div
      class="circuit-0 circuit-1"
      width="100%"
    />
  </div>
</div>
`;

exports[`Slide styles should render with forward animation styles 1`] = `
@keyframes animation-0 {
  from {
    width: 0%;
  }

  to {
    width: 100%;
  }
}

.circuit-4 {
  width: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 800px;
  -ms-flex-preferred-size: 800px;
  flex-basis: 800px;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  position: relative;
  z-index: 2;
}

.circuit-2 {
  box-shadow: 0 0 1px rgba(0,0,0,0.05);
  overflow: hidden;
  will-change: width;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-duration: 640ms;
  animation-duration: 640ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.circuit-0 {
  width: 800px;
}

<div
  class="circuit-4 circuit-5"
  width="800px"
>
  <div
    class="circuit-2 circuit-3"
  >
    <div
      class="circuit-0 circuit-1"
      width="800px"
    />
  </div>
</div>
`;
